src/NuGet.Core/NuGet.Packaging/Rules/MissingReadmeRule.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+
+namespace NuGet.Packaging.Rules
+{
+    public class MissingReadmeRule : IPackageRule
+    {
+        public string MessageFormat { get; }
+
+        public MissingReadmeRule(string messageFormat)
+        {
+            MessageFormat = messageFormat ?? throw new ArgumentNullException(nameof(messageFormat));
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            var nuspecReader = builder?.NuspecReader;
+            var readmeMetadata = nuspecReader?.GetReadme();
+
+            if (readmeMetadata == null)
+            {
+                yield return PackagingLogMessage.CreateMessage(
+                    string.Format(CultureInfo.CurrentCulture, Environment.NewLine + MessageFormat),
```

Comment:
Originalyl I was wondering if this was raising a warning, but now I'm not clear. 

@heng-liu can we get a more detailed explanation of what we're implementing here? The issues says the first part, but it doesn't go in detail.

Reply:
Thanks for reviewing @zivkan !
I tried without newline at first, the output is as below, and the message seems very difficult to discover.
Without newline:
```
C:\PatchedSDK1\dotnet.exe pack C:\Users\username\Repos\Classlib1\Classlib1.csproj
MSBuild version 17.5.0-preview-23061-01+040e2a90e for .NET
  Determining projects to restore...
  Restored C:\Users\username\Repos\Classlib1\Classlib1.csproj (in 458 ms).
  Classlib1 -> C:\Users\username\Repos\Classlib1\bin\Debug\net7.0\Classlib1.dll
  The created package is missing a readme. Go to https://learn.microsoft.com/en-us/nuget/create-packages/package-author
  ing-best-practices#readme to learn why package readmes are important.
  Successfully created package 'C:\Users\username\Repos\Classlib1\bin\Debug\Classlib1.1.0.0.nupkg'.
```
With newline:
```
C:\PatchedSDK2\dotnet.exe pack C:\Users\username\Repos\Classlib1\Classlib1.csproj
MSBuild version 17.5.0-preview-23061-01+040e2a90e for .NET
  Determining projects to restore...
  Restored C:\Users\username\Repos\Classlib1\Classlib1.csproj (in 549 ms).
  Classlib1 -> C:\Users\username\Repos\Classlib1\bin\Debug\net7.0\Classlib1.dll

  The created package is missing a readme. Go to https://learn.microsoft.com/en-us/nuget/create-packages/package-author
  ing-best-practices#readme to learn why package readmes are important.
  Successfully created package 'C:\Users\username\Repos\Classlib1\bin\Debug\Classlib1.1.0.0.nupkg'.
```
Hi @JonDouglas @albarry4 , may I know what do you think? Thanks!

Reply:
@heng-liu that's basically just how MSBuild works. Do we want to "work against" the normal for this one rule, or stay consistent with everything else?

Also try a single project with multiple pack warnings, not just one.  Also try multiple projects in the solution (each of which have multiple warnings) and pack the solution. Basically, when trying to optimise the output, look at as many different scenarios as possible, not just one.

Reply:
Also, can you please show sample output when running `dotnet pack my.sln`, where there are two projects.  I'm used to MSBuild messages saying which project file (or in the case of C#, which code file) the message is related to.

If my solution has multiple projects that get packed, is the message clear about which project it's referring to?

Reply:
> @heng-liu that's basically just how MSBuild works. Do we want to "work against" the normal for this one rule, or stay consistent with everything else?
> 
> Also try a single project with multiple pack warnings, not just one. Also try multiple projects in the solution (each of which have multiple warnings) and pack the solution. Basically, when trying to optimise the output, look at as many different scenarios as possible, not just one.

Thanks for the explanation! Removed the newline in the message.

Reply:
> Originalyl I was wondering if this was raising a warning, but now I'm not clear.
> 
> @heng-liu can we get a more detailed explanation of what we're implementing here? The issues says the first part, but it doesn't go in detail.

The [CreateMessage](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/Errors/PackagingLogMessage.cs#L70) doesn't set `logLevel `to `LogLevel.Warning` nor `LogLevel.Error` so it's not a warning nor an error. I set it to `LogLevel.Minimal`, which is the same with the messages [Log_PackageCommandSuccess](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs#L202-L204)
@zivkan also mentioned it needs minimal verbosity as `dotnet pack` defaults to minimal in [this comment](https://github.com/NuGet/Home/issues/12070#issuecomment-1242520006) .
BTW, the loglevel mapping could be found [here](https://github.com/NuGet/Home/blob/dev/implemented/2021/Verify-Verbosity.md#log-level-mapping---the-following-details-are-copied-from-here-thanks-to-joel-verhagen-for-the-detailed-information)

Reply:
> Also try a single project with multiple pack warnings, not just one. Also try multiple projects in the solution (each of which have multiple warnings) and pack the solution. Basically, when trying to optimise the output, look at as many different scenarios as possible, not just one.

Tried a solution with 4 projects: ClassLibrary1(missing readme), ClassLibrary2, ClassLibrary3(missing readme), ClassLibrary4. 
```
C:\PatchedSDK0301\dotnet.exe pack C:\Users\Source\Repos\ClassLibrary1\ClassLibrary1.sln
MSBuild version 17.5.0-preview-23061-01+040e2a90e for .NET
  Determining projects to restore...
  Restored C:\Users\Source\Repos\ClassLibrary1\ClassLibrary3\ClassLibrary3.csproj (in 1.7 sec).
  Restored C:\Users\Source\Repos\ClassLibrary1\ClassLibrary1\ClassLibrary1.csproj (in 1.7 sec).
  Restored C:\Users\Source\Repos\ClassLibrary1\ClassLibrary2\ClassLibrary2.csproj (in 1.7 sec).
  Restored C:\Users\Source\Repos\ClassLibrary1\ClassLibrary4\ClassLibrary4.csproj (in 1.7 sec).
  ClassLibrary2 -> C:\Users\Source\Repos\ClassLibrary1\ClassLibrary2\bin\Debug\net7.0\ClassLibrary2.dll
  ClassLibrary1 -> C:\Users\Source\Repos\ClassLibrary1\ClassLibrary1\bin\Debug\net7.0\ClassLibrary1.dll
  ClassLibrary3 -> C:\Users\Source\Repos\ClassLibrary1\ClassLibrary3\bin\Debug\net7.0\ClassLibrary3.dll
  ClassLibrary4 -> C:\Users\Source\Repos\ClassLibrary1\ClassLibrary4\bin\Debug\net7.0\ClassLibrary4.dll
  Successfully created package 'C:\Users\Source\Repos\ClassLibrary1\ClassLibrary2\bin\Debug\ClassLibrary2.1.0.0.nupkg'.
  The created package is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
  Successfully created package 'C:\Users\Source\Repos\ClassLibrary1\ClassLibrary4\bin\Debug\ClassLibrary4.1.0.0.nupkg'.
  Successfully created package 'C:\Users\Source\Repos\ClassLibrary1\ClassLibrary3\bin\Debug\ClassLibrary3.1.0.0.nupkg'.
  The created package is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
  Successfully created package 'C:\Users\Source\Repos\ClassLibrary1\ClassLibrary1\bin\Debug\ClassLibrary1.1.0.0.nupkg'.
```

Reply:
I don't know if other pack warnings/errors have the same problem, but I think this sample output shows a problem, because the output makes it look like either ClassLibrary2 and ClassLibrary3 have issues, or maybe ClassLibrary4 and ClassLibrary1 (it's not clear), but your description says that the messages are related to ClassLibrary1 and ClassLibrary3, but there's no way for us to know that looking at the output.

Please check a different warning type (I don't know, iconURL is deprecated?). If it has the same problem, then please create a new issue and we can ignore it here. Otherwise, I think we need to improve this message's output.